python/test/test_polars.py

```diff
@@ -548,3 +548,69 @@ def test_replace_binary_path():
     assert str(m_expr((pl.LazyFrame(dict()), pl.all()))) == "len().testing!:noise_plugin()"
 
     del os.environ["OPENDP_LIB_PATH"]
+
+
+def test_pickle_bomb():
+    pl = pytest.importorskip("polars")
+
+    from polars._utils.parse import parse_into_list_of_expressions  # type: ignore[import-not-found]
+    from opendp._lib import lib_path
+    import io
+    import pickle
+
+    poison_binary = b'c__builtin__\neval\n(V1 / 0\ntR.'
+
+    # poison_binary raises a ZeroDivisionError if it is ever unpickled
+    with pytest.raises(ZeroDivisionError):
+        pickle.loads(poison_binary)
+
+    # craft an expression that contains a poisoned pickle binary
+    from polars._utils.wrap import wrap_expr
+
+    pyexprs = parse_into_list_of_expressions((pl.len(), pl.lit("Laplace"), pl.lit(0.7)))
+    bomb_expr = wrap_expr(
+        pl.polars.register_plugin_function(
+            plugin_path=str(lib_path),
+            function_name="noise",
+            args=pyexprs,
+            kwargs=poison_binary,
+            is_elementwise=True,
+            input_wildcard_expansion=False,
+            returns_scalar=False,
+            cast_to_supertype=False,
+            pass_name_to_apply=False,
+            changes_length=False,
+        )
+    )
+
+    # craft a lazyframe that contains a poisoned pickle binary
+    lf_domain, lf = example_lf()
+    bomb_lf = lf.select(bomb_expr)
+
+    # ensure that ser/de round-trip of expression does not trigger pickle
+    ser_expr = bomb_expr.meta.serialize()
+    pl.Expr.deserialize(io.BytesIO(ser_expr))
+
+    # ensure that ser/de round-trip of lazyframe does not trigger pickle
+    ser_lf = bomb_lf.serialize()
+    pl.LazyFrame.deserialize(io.BytesIO(ser_lf))
+
+    # OpenDP explicitly rejects any pickled data it finds
+    err_msg = "OpenDP does not allow pickled keyword arguments as they may enable remote code execution."
+    with pytest.raises(dp.OpenDPException) as err:
+        dp.m.make_private_expr(
+            dp.expr_domain(lf_domain, grouping_columns=[]),
+            dp.partition_distance(dp.symmetric_distance()),
+            dp.max_divergence(T=float),
+            bomb_expr,
+        )
+    assert err.value.message == err_msg
+
+    with pytest.raises(dp.OpenDPException) as err:
+        dp.m.make_private_lazyframe(
+            lf_domain,
+            dp.symmetric_distance(),
+            dp.max_divergence(T=float),
+            bomb_lf,
+        )
+    assert err.value.message == err_msg
```

Comment:
use the `match` kwarg?
```suggestion
    err_msg_re = re.escape("OpenDP does not allow pickled keyword arguments as they may enable remote code execution.")
    with pytest.raises(dp.OpenDPException, match=err_msg_re) as err:
        dp.m.make_private_expr(
            dp.expr_domain(lf_domain, grouping_columns=[]),
            dp.partition_distance(dp.symmetric_distance()),
            dp.max_divergence(T=float),
            bomb_expr,
        )

    with pytest.raises(dp.OpenDPException, match=err_msg_re) as err:
        dp.m.make_private_lazyframe(
            lf_domain,
            dp.symmetric_distance(),
            dp.max_divergence(T=float),
            bomb_lf,
        )
```